### ISSUE TITLE
walking progress messages

### ISSUE THREAD

**Issue by gonsie** [hpc/mpifileutils#296]
For dwalk / any tool that does a walking step. This was reported by @ofaaland using drm.

If the file hierarchy is made up of many empty directories, the Walking progress messages could make it appear as if the tool is hung. This is due to the fact that it is fast to discover the number of empty dirs, but the subsequent `getdents` call on each dir takes a while. 

## Example output:

> [2019-07-10T09:17:37] Walking /p/.../dir                                                          
[2019-07-10T09:17:51] Items walked 63921                                                    
[2019-07-10T09:18:01] Items walked 63921                                                    
[2019-07-10T09:18:11] Items walked 63921                                                    
[2019-07-10T09:18:22] Items walked 63921                                                    
[2019-07-10T09:18:32] Items walked 63921                                                    
[2019-07-10T09:18:42] Items walked 63921                                                    
[2019-07-10T09:20:55] Items walked 63921
[2019-07-10T09:21:04] Items walked 63921
[2019-07-10T09:21:05] Walked 63921 items in 207.115070 seconds (308.625539 files/sec)

## Requested output:

Something along the lines of:

> Items walked 63921
> Examining items 0 / 63921
> Examining items 1000 / 63921
> Examining items 5500 / 63921

**Comment by adammoody** [comment]
These progress messages are coming from our libcircle reduction callbacks here.

https://github.com/hpc/mpifileutils/blob/2de9b3076cc7738eeae68a3d0000f408620bc308/src/common/mfu_flist_walk.c#L80

We could look at adding a second counter, one to track number of items discovered, and another to track number of items completed.  For files and symlinks, we'd bump both counters at once.  For a directory, we could increment the discovered counter right away, and then increment the completed counter after we call closedir().

**Comment by gonsie** [comment]
That would be good, maybe something along the lines of:

> Items walked x / n

Where x is the number of item's examined, and n is the number of items discovered. I think users would understand if both x and n are increasing during the walk phase. It would also indicate that the walk phase is over once x == n.

**Comment by adammoody** [comment]
Merged https://github.com/hpc/mpifileutils/pull/297 to increment the reduction counter after processing a directory rather than upon first discovering it.  This way we don't count the directory until we are fully done with it.  We could still add a second counter to track discovered-but-not-yet-processed items later as we first discussed here.  However, since we have a fix for the original report, I'll close this issue out.

Thanks for reporting the issue, @ofaaland !
